### ISSUE TITLE
add support for pre-existing gateway nodes

### ISSUE THREAD

**Issue by raffaelespazzoli** [open-cluster-management/submariner-addon#84]
sometime people will want to have full control on how to create the gateway node. If deploying via Hive, it's very easy to declaratively add a new machineset with full control on the machine characteristics, for example. It should be possible to specify that the gateway not should not be created and perhaps specify a node selector to select the gatewya nodes. If the selector if fixed (I believe it's hard coded to be `submariner.io/gateway: 'true'`) then the system might just check to see if nodes like that exists and create new ones only if they don't.

**Comment by sridhargaddam** [comment]
Necessary support is added in cloud-prepare for GCP clusters via the following PR - https://github.com/submariner-io/cloud-prepare/issues/97
It needs to be consumed from the submariner add-on - this is currently in progress.

**Comment by nyechiel** [comment]
@aswinsuryan this is related to our discussion on this topic. We agreed that with the exception of AWS - we need to support both dedicated and existing/non-dedicated nodes (with default being non-dedicated). There is also this related issue on the console side: https://github.com/stolostron/backlog/issues/19296

Can you own this for 2.5?

**Comment by aswinsuryan** [comment]
> @aswinsuryan this is related to our discussion on this topic. We agreed that with the exception of AWS - we need to support both dedicated and existing/non-dedicated nodes (with default being non-dedicated). There is also this related issue on the console side: [stolostron/backlog#19296](https://github.com/stolostron/backlog/issues/19296)
> 
> Can you own this for 2.5?

Yes I can, for the supported Clouds, GCP , RHOS and Azure(if can be supported there)

**Comment by nyechiel** [comment]
@sridhargaddam @aswinsuryan can you please add a comment explaining why and close it? (tl;dr available with `subctl`, not with ACM)

**Comment by sridhargaddam** [comment]
From an upstream POV, we support both dedicated as well as non-dedicated gateway nodes (i.e., use one of the existing nodes which already has the required label and if there is none, cloud prepare will label one/more of the worker nodes as GW nodes). However, in ACM we are only supporting a dedicated gateway node as we wanted to avoid making out-of-band changes to the underlying infra since there is a possibility that the changes can get overridden later on. Also, creating a dedicated gateway node allows us to choose (or provide an option to the user) an appropriate instance type for the Gateway node.

**Comment by sridhargaddam** [comment]
@nyechiel I do not see an option to close this issue.
